service/controller/clusterapi/v29/machine_deployment_resource_set.go

```diff
@@ -76,6 +139,7 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 	resources := []controller.Resource{
 		awsClientResource,
 		encryptionResource,
+		ipamResource,
```

Comment:
I'd maybe move this below that `machineDeploymentSubnetResource` so that we retain existing functionality until we can let that go (basically after AZ/Subnet mapping refactoring PRs are merged & in use). Currently due to TCCP CF structure, the worker ASGs subnets must be from the TCCP and this would break it. When my open PRs are merged, that's not used anymore and `MachineDeployment` gets more independent (and at that point we can remove that resource and this IPAM resource takes effect.